fem/integ/bilininteg_diffusion_kernels.cpp

```diff
@@ -14,6 +14,33 @@
 namespace mfem
 {
 
+// PA Diffusion Integrator
+
+DiffusionIntegrator::Kernels DiffusionIntegrator::kernels;
+DiffusionIntegrator::Kernels::Kernels()
+{
+   // 2D
+   DiffusionIntegrator::AddSpecialization2D<2,2>();
+   DiffusionIntegrator::AddSpecialization2D<3,3>();
+   DiffusionIntegrator::AddSpecialization2D<4,4>();
+   DiffusionIntegrator::AddSpecialization2D<5,5>();
+   DiffusionIntegrator::AddSpecialization2D<6,6>();
+   DiffusionIntegrator::AddSpecialization2D<7,7>();
+   DiffusionIntegrator::AddSpecialization2D<8,8>();
+   DiffusionIntegrator::AddSpecialization2D<9,9>();
+   // 3D
+   DiffusionIntegrator::AddSpecialization3D<2,2>();
+   DiffusionIntegrator::AddSpecialization3D<2,3>();
+   DiffusionIntegrator::AddSpecialization3D<3,4>();
+   DiffusionIntegrator::AddSpecialization3D<4,5>();
+   DiffusionIntegrator::AddSpecialization3D<4,6>();
+   DiffusionIntegrator::AddSpecialization3D<5,6>();
+   DiffusionIntegrator::AddSpecialization3D<5,8>();
+   DiffusionIntegrator::AddSpecialization3D<6,7>();
+   DiffusionIntegrator::AddSpecialization3D<7,8>();
+   DiffusionIntegrator::AddSpecialization3D<8,9>();
+}
+
```

Comment:
Thanks for the comment--these are hardcoded to demonstrate the analogue between this method and the switch based dispatch.  Whether we want to include the same specializations using this method or remove them altogether I think is up for debate.  If we eliminate these lines as well as the switch statements, this will need to be well documented so that users don't experience a massive performance regression.  Thoughts @tzanio ?

Reply:
As a first step I suggest we keep those and not change the behavior yet. We can do that in a later PR and document appropriately.